iroha_data_model/src/lib.rs

```diff
@@ -38,6 +38,27 @@ pub type Name = String;
 /// Represents a sequence of bytes. Used for storing encoded data.
 pub type Bytes = Vec<u8>;
 
+/// Collection of parameters by their names.
+pub type Metadata = BTreeMap<Name, Value>;
+
+/// Similar to [`std::convert::AsMut`] but indicating that this reference conversion can fail.
+pub trait TryAsMut<T> {
+    /// The type returned in the event of a conversion error.
+    type Error;
+
+    /// Performs the conversion.
+    fn try_as_mut(&mut self) -> StdResult<&mut T, Self::Error>;
```

Comment:
Actually `Err` variant in `iroha_error::Result` can be overrided by different error. By default it is `iroha_error::Error`. Probably can be changed to regular result.